fs/sync/sync_test.go

```diff
@@ -1202,6 +1202,93 @@ func TestSyncOverlap(t *testing.T) {
 	checkErr(Sync(context.Background(), FremoteSync, FremoteSync, false))
 }
 
+// Test with CompareDest set
+func TestSyncCompareDest(t *testing.T) {
+	r := fstest.NewRun(t)
+	defer r.Finalise()
+
+	fs.Config.CompareDest = r.FremoteName + "/CompareDest"
+	defer func() {
+		fs.Config.CompareDest = ""
+	}()
+
+	file2 := r.WriteObject(context.Background(), "CompareDest/one", "one", t1)
```

Comment:
I think this probably needs more cases in it for the destination object
- destination object exists in destination dir
- destination object exists in CompareDest
- destination object exists in dir and CompareDest
- destination object does not exist